kernel/memfs/src/lib.rs

```diff
@@ -36,7 +36,7 @@ pub struct MemFile {
 
 impl MemFile {
     /// Allocates writable memory space for the given `contents` and creates a new file containing that content in the given `parent` directory.
-    pub fn new(name: String, parent: &DirRef) -> Result<FileRef, &'static str> {
+    pub fn new_fileref(name: String, parent: &DirRef) -> Result<FileRef, &'static str> {
```

Comment:
Right, clippy wants `new` to always return `Self`, so you renamed it here.
I agree, maybe someone else would like to bikeshed on the naming.